### ISSUE TITLE
We should probably use utf8 everywhere to avoid encoding errors.

### ISSUE THREAD

**Issue by HatuneMiku** [RasaHQ/rasa#8199]
**Rasa version**:
2.4
**Rasa SDK version** (if used & relevant):

**Rasa X version** (if used & relevant):

**Python version**:

**Operating system** (windows, osx, ...):
Windows 10
**Issue**:
If Windows10 system language is Chinese, read yaml file raise encoding error. My nlu.yml include some chinses char. 

For this error, just change:
https://github.com/RasaHQ/rasa/blob/2ed20f6239e181788df98159c7470122887b83af/rasa/shared/core/training_data/story_reader/yaml_story_reader.py#L182
TO:
```python
    with open(file_path, encoding="utf8") as file: 
```
Could we use utf8 every please. like every functions about `load` and `save` files.  Thanks!
 

**Error (including full traceback)**:
```
Traceback (most recent call last):
  File "D:\Programs\Python\Python37\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "D:\Programs\Python\Python37\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "D:\Programs\Python\venv\rasa2\Scripts\rasa.exe\__main__.py", line 7, in <module>    
  File "d:\programs\python\venv\rasa2\lib\site-packages\rasa\__main__.py", line 116, in main
    cmdline_arguments.func(cmdline_arguments)
  File "d:\programs\python\venv\rasa2\lib\site-packages\rasa\cli\train.py", line 207, in train_nlu
    finetuning_epoch_fraction=args.epoch_fraction,
  File "d:\programs\python\venv\rasa2\lib\site-packages\rasa\train.py", line 711, in train_nlu
    finetuning_epoch_fraction=finetuning_epoch_fraction,
  File "d:\programs\python\venv\rasa2\lib\site-packages\rasa\utils\common.py", line 307, in run_in_loop
    result = loop.run_until_complete(f)
  File "D:\Programs\Python\Python37\lib\asyncio\base_events.py", line 587, in run_until_complete
    return future.result()
  File "d:\programs\python\venv\rasa2\lib\site-packages\rasa\train.py", line 738, in train_nlu_async
    config, domain, training_data_paths=[nlu_data]
  File "d:\programs\python\venv\rasa2\lib\site-packages\rasa\shared\importers\importer.py", line 117, in load_nlu_importer_from_config
    config_path, domain_path, training_data_paths, TrainingType.NLU
  File "d:\programs\python\venv\rasa2\lib\site-packages\rasa\shared\importers\importer.py", line 86, in load_from_config
    config, config_path, domain_path, training_data_paths, training_type
  File "d:\programs\python\venv\rasa2\lib\site-packages\rasa\shared\importers\importer.py", line 151, in load_from_dict
    config_path, domain_path, training_data_paths, training_type
  File "d:\programs\python\venv\rasa2\lib\site-packages\rasa\shared\importers\rasa.py", line 34, in __init__
    training_data_paths, rasa.shared.data.is_story_file
  File "d:\programs\python\venv\rasa2\lib\site-packages\rasa\shared\data.py", line 152, in get_data_files
    new_data_files = _find_data_files_in_directory(path, filter_predicate)
  File "d:\programs\python\venv\rasa2\lib\site-packages\rasa\shared\data.py", line 172, in _find_data_files_in_directory
    if filter_property(full_path):
  File "d:\programs\python\venv\rasa2\lib\site-packages\rasa\shared\data.py", line 221, in is_story_file
    file_path
  File "d:\programs\python\venv\rasa2\lib\site-packages\rasa\shared\core\training_data\story_reader\yaml_story_reader.py", line 164, in is_stories_file
    file_path, KEY_STORIES, KEY_RULES
  File "d:\programs\python\venv\rasa2\lib\site-packages\rasa\shared\core\training_data\story_reader\yaml_story_reader.py", line 185, in is_key_in_yaml
    for line in file
  File "d:\programs\python\venv\rasa2\lib\site-packages\rasa\shared\core\training_data\story_reader\yaml_story_reader.py", line 184, in <genexpr>
    any(line.lstrip().startswith(f"{key}:") for key in keys)
UnicodeDecodeError: 'gbk' codec can't decode byte 0x80 in position 94: illegal multibyte sequence
```

**Command or request that led to error**:
```
rasa train nlu
```

**Comment by sara-tagger** [comment]
Thanks for raising this issue, @amn41 will get back to you about it soon✨

###### Please also check out the [docs](https://rasa.com/docs/) and the [forum](https://forum.rasa.com/) in case your issue was raised there too 🤗

**Comment by HatuneMiku** [comment]
rasa test output `intent_errors.json` output chars do not display the actual characters but the encoding. maybe we should add `ensure_ascii=False` somewhere.

**Comment by HatuneMiku** [comment]
change  line 495

https://github.com/RasaHQ/rasa/blob/4b8f7459b1a234bccaf999a56510c99aa93221ee/rasa/shared/utils/io.py#L492-L495

to

```python
    write_text_file(json.dumps(obj, ensure_ascii=False, indent=2), filename) 
```

**Comment by HatuneMiku** [comment]
> change line 495
> 
> https://github.com/RasaHQ/rasa/blob/4b8f7459b1a234bccaf999a56510c99aa93221ee/rasa/shared/utils/io.py#L492-L495
> 
> to
> 
> ```python
>     write_text_file(json.dumps(obj, ensure_ascii=False, indent=2), filename) 
> ```

https://github.com/RasaHQ/rasa/pull/8308

**Comment by HatuneMiku** [comment]
The encoding error seems to be fixed in version 2.4.2.
